Source/Core/Core/Scripting/ScriptUtilities.cpp

```diff
@@ -0,0 +1,764 @@
+    if (current_event.event_type == ScriptQueueEventTypes::CreateScriptFromUI)
+    {
+      InitializeScript(current_event.unique_script_identifier, current_event.script_filename,
+                       current_event.print_callback_func, current_event.script_end_callback_func);
+    }
+    else if (current_event.event_type == ScriptQueueEventTypes::StopScriptFromUI ||
+             current_event.event_type == ScriptQueueEventTypes::StopScriptFromScriptEndCallback)
+    {
+      StopScript(current_event.unique_script_identifier);
+    }
+    else
+    {
+      queuePanicAlertEvent("Error: Encountered unknown ScriptQueueEventType in "
+                           "ProcessScriptQueueEvents() function.");
+      return;
+    }
+    script_events.erase(script_events.begin());
+  }
+}
+
+bool StartScripts()
+{
+  std::lock_guard<std::mutex> lock(initialization_and_destruction_lock);
+  std::lock_guard<std::mutex> second_lock(global_code_and_frame_callback_running_lock);
+  bool return_value = false;
+  if (list_of_all_scripts.size() == 0 || queue_of_scripts_waiting_to_start.Empty())
+    return false;
+  while (!queue_of_scripts_waiting_to_start.Empty())
+  {
+    ScriptContext* next_script = nullptr;
+    queue_of_scripts_waiting_to_start.Pop(next_script);
+    if (next_script == nullptr)
+      break;
+    next_script->script_specific_lock.lock();
+    if (next_script->is_script_active)
+    {
+      next_script->current_script_call_location = ScriptCallLocationsEnum::FromScriptStartup;
+      next_script->dll_specific_api_definitions.StartScript(next_script);
+      return_value = next_script->called_yielding_function_in_last_global_script_resume;
+      if (next_script->script_return_code != 0)
+        queuePanicAlertEvent(next_script->last_script_error);
+    }
+    next_script->script_specific_lock.unlock();
+    if (return_value)
+      break;
+  }
+  return return_value;
+}
+
+bool RunGlobalCode()
+{
+  std::lock_guard<std::mutex> lock(global_code_and_frame_callback_running_lock);
+  bool return_value = false;
+  for (size_t i = 0; i < list_of_all_scripts.size(); ++i)
+  {
+    ScriptContext* current_script = list_of_all_scripts[i];
+    current_script->script_specific_lock.lock();
+    if (current_script->is_script_active && !current_script->finished_with_global_code)
+    {
+      current_script->current_script_call_location =
+          ScriptCallLocationsEnum::FromFrameStartGlobalScope;
+      current_script->dll_specific_api_definitions.RunGlobalScopeCode(current_script);
+      return_value = current_script->called_yielding_function_in_last_global_script_resume;
+      if (current_script->script_return_code != 0)
+        queuePanicAlertEvent(current_script->last_script_error);
+    }
+    current_script->script_specific_lock.unlock();
+    if (return_value)
+      break;
+  }
+  return return_value;
+}
+
+bool RunOnFrameStartCallbacks()
+{
+  std::lock_guard<std::mutex> lock(global_code_and_frame_callback_running_lock);
```

Comment:
```suggestion
  std::lock_guard lock(global_code_and_frame_callback_running_lock);
```

ditto for other locks in the file